File: 2022-2023/sourcecode/lab/lab5/CarsyAppJava/app/src/main/java/pl/udu/uwr/pum/carsyappjava/adapters/OverviewCarsAdapter.java
package pl.udu.uwr.pum.carsyappjava.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import pl.udu.uwr.pum.carsyappjava.R;
import pl.udu.uwr.pum.carsyappjava.data.Car;
import pl.udu.uwr.pum.carsyappjava.data.Cost;
import pl.udu.uwr.pum.carsyappjava.data.DataProvider;
import pl.udu.uwr.pum.carsyappjava.util.FormatterUtil;

public class OverviewCarsAdapter extends RecyclerView.Adapter<OverviewCarsAdapter.ViewHolder> {

    private final ArrayList<Car> cars = DataProvider.getCars();

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private final TextView carNameTextView;
        private final TextView brandTextView;
        private final TextView modelTextView;
        private final TextView yearTextView;
        private final TextView totalCostsTextView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            carNameTextView= itemView.findViewById(R.id.overviewViewPagerCarName);
            brandTextView = itemView.findViewById(R.id.overviewViewPagerBrand);
            modelTextView = itemView.findViewById(R.id.overviewViewPagerModel);
            yearTextView = itemView.findViewById(R.id.overviewViewPagerYearOfProduction);
            totalCostsTextView = itemView.findViewById(R.id.overviewViewPagerTotalCosts);
        }

        public void bind (Car item){
            carNameTextView.setText(item.getName());
            brandTextView.setText(item.getBrand());
            modelTextView.setText(item.getModel());
            yearTextView.setText(item.getYearOfProduction());
            totalCostsTextView.setText(String.format("%s zł",
                    FormatterUtil.decimalFormat.format(
                            item.getCosts().stream()
                                    .map(Cost::getAmount)
                                    .reduce(0, Integer::sum))));
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(
                R.layout.item_recyclerview_cars_overview, parent, false
        ));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Car item = cars.get(position);
    }

    @Override
    public int getItemCount() {
        return cars.size();
    }
}

File: 2022-2023/sourcecode/lab/lab5/CarsyAppKotlin/app/src/main/java/pl/udu/uwr/pum/carsyappkotlin/adapters/OverviewCarsAdapter.kt
package pl.udu.uwr.pum.carsyappkotlin.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import pl.udu.uwr.pum.carsyappkotlin.R
import pl.udu.uwr.pum.carsyappkotlin.data.Car
import pl.udu.uwr.pum.carsyappkotlin.data.DataProvider
import pl.udu.uwr.pum.carsyappkotlin.util.decimalFormat

class OverviewCarsAdapter : RecyclerView.Adapter<OverviewCarsAdapter.ViewHolder>() {

    private val cars = DataProvider.cars

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        private val carNameTextView: TextView = view.findViewById(R.id.overviewViewPagerCarName)
        private val brandTextView: TextView = view.findViewById(R.id.overviewViewPagerBrand)
        private val modelTextView: TextView = view.findViewById(R.id.overviewViewPagerModel)
        private val yearTextView: TextView = view.findViewById(R.id.overviewViewPagerYearOfProduction)
        private val totalCostsTextView: TextView = view.findViewById(R.id.overviewViewPagerTotalCosts)

        fun bind (item: Car){
            carNameTextView.text = item.name
            brandTextView.text = item.brand
            modelTextView.text = item.model
            yearTextView.text = item.yearOfProduction.toString()
            totalCostsTextView.text = decimalFormat.format(item.costs.sumOf { it.amount }).toString()
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context).inflate(
            R.layout.item_recyclerview_cars_overview, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = cars[position]
        holder.bind(item)
    }

    override fun getItemCount(): Int = cars.size
}